.github/workflows/pypi-release.yml

```diff
@@ -1,25 +1,45 @@
 name: Publish to PyPI
 on:
   push:
-    tags:
-      - "*"
+    branches: [main]  # this will just build and check package
+    tags: ["*"]  # this will run full workflow including publish to PyPI
+  pull_request:
+    branches: [main]  # this will just build and check package
+
 jobs:
-  publish:
+  build:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
       - name: Set up Python
         uses: actions/setup-python@v4
         with:
           python-version: '3.x'
+
       - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install setuptools wheel twine
-      - name: Build and publish
-        env:
-          TWINE_USERNAME: __token__
-          TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
-        run: |
-          python setup.py sdist bdist_wheel
-          twine upload dist/*
+        run: pip install -U build twine
+      - name: Build 📦 package
+        run: python -m build
+      - name: Check 📦 package
+        run: twine check dist/*
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: package
+          path: dist
+
+  publish:
+    # only run this for tags
+    if: startsWith(github.event.ref, 'refs/tags/')
+    needs: [build]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/download-artifact@v3
+        with:
+          name: package
+          path: dist
+      - name: Publish 📦 to PyPI
+        uses: pypa/gh-action-pypi-publish@v1.8.7
```

Comment:
Yes, this simplifies our code, but pulls in another action that does not add any features. It merely calls `twine upload dist/*`. I have reviewed that action and I do not like the code base. I don't want to rely on that action and pass the pypi password to it for no benefit. I am fine with setting up python and installing twine in our workflow.

Reply:
It is official pypi action...

Reply:
```suggestion
        run: |
          pip install -q twine
          twine upload dist/*
```